src/raftstore/store/peer.rs

```diff
@@ -631,9 +637,19 @@ impl Peer {
         self.get_store().applied_index_term == self.term()
     }
 
+    pub fn schedule_apply_proposals(&mut self, worker: &Worker<ApplyTask>) {
+        if self.apply_proposals.is_empty() {
+            return;
+        }
+
+        let proposals = mem::replace(&mut self.apply_proposals, vec![]);
```

Comment:
`Vec` will allocate 4 or more buffer once an item is inserted. And if it's a dormant region, the pre-allocated memory is wasted.